src/utils/kmlUtils.js

```diff
@@ -87,4 +84,395 @@ export function getKmlExtentForProjection(projection, extent) {
     return null
 }
 
+/**
+ * Get the KML feature type
+ *
+ * The type is taken from the geoadmin proprietary "type" property, and if this property is not
+ * available it is guess from the geometry and style. The only use case that cannot be guess is for
+ * measure which will always fallback to line.
+ *
+ * @param {ol/Feature} kmlFeature Open layer kml feature
+ * @returns {String | null} KML feature type or null if this is not a geoadmin kml feature
+ */
+export function getFeatureType(kmlFeature) {
+    let featureType = kmlFeature.get('type')?.toUpperCase() // only set by geoadmin's kml
+    if (!Object.values(EditableFeatureTypes).includes(featureType)) {
+        log.error(
```

Comment:
Actually here this is temporary until we do the rewrite with the GeometryFeature (see PB-269). I think once we can have another generic feature type to handle external kml, this function should only return the type from our special tag and not try to guess it. Then the current `parseKml` function in his util should only be done for our kml and we would need a new parseKml or modify this once to return GeometryFeatures instead of EditableFeatures for external kml. So I propose to let it be so for the moment.